### ISSUE TITLE
Hiding labels do not work as expected

### ISSUE THREAD

**Issue by nknguyenhc** [CATcher-org/WATcher#240]
**Describe the bug**
When hiding a label, from the dropdown on the top right corner, only the labels are hidden, but not the related issues & PRs.

**To Reproduce**

1. Open any repository with active labels.
2. Click on hiding any label.

**Expected behavior**
It should hide the issues & PRs that have the label. The current behaviour is that only the label becomes hidden, but not the related issues & PRs.

I think it makes more sense that when a user clicks on hiding a label, the user wants to hide the relevant issues & PRs with the label.

**Screenshots**

![image](https://github.com/CATcher-org/WATcher/assets/87511888/884a8fb6-b22b-40fb-858e-f1f3d68cd6ec)

After hiding the label `difficulty.Easy`:

![image](https://github.com/CATcher-org/WATcher/assets/87511888/b4d39df5-bab7-457b-a788-d945ad904668)

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser Chrome
- Version 1.1.0

**Additional context**
As discussed in #230

**Comment by nknguyenhc** [comment]
What do you guys think? @CATcher-org/2324s2

**Comment by gycgabriel** [comment]
This is certainly something worth thinking about. 

Is there a use case for hiding the UI label without excluding it from search? If there is, 
- How can we present this feature in a better way
- Would it be too much effort for the user to click through each label that they are not interested in just to hide a UI

Or, 
- What are some alternative to minimize clutter from the labels?
Unlike Github, WATcher displays the issues in shorter width, so it is much easier to run out of horizontal space to display all the labels

> Three-state checkbox

Kind of goes against what users expect of a checkbox, but it is a possible solution

I would suggest together with that, also change the label colours or some bigger indicator on the label name itself so it is more obvious to the user if it is negated or selected. As the checkbox is quite small

**Comment by nknguyenhc** [comment]
I would agree with the 3-state checkbox as a solution to incorporate the feature of hiding relevant issues & PRs. Perhaps instead of presenting it as a checkbox, we can indicate green border around the label in the dropdown when filtering the label in (selected), and red when filtering the label out (negated)? I think by incorporating this 3-state button, we can also better present hiding just the label better as a feature than a bug.

For the feature of hiding just the label to reduce clutter, I think the feature is fine as it is right now. It is sometimes necessary to reduce the clutter when there are too many labels used on an issue. When the user wants to unhide the label, the user would still have to go back to the list to click on the eye icon. However, it would still be too much effort for the user to go through a long list of labels just to hide a label from the table. Perhaps a possible enhancement to this feature is to have an eye icon on the label on each issue/PR card, which the user can just click to hide the label from the table?

**Comment by nknguyenhc** [comment]
First part of 3-state labels has been done. Second part to this is to create an "eye icon" besides each label, so that the user can click on the eye icon to hide the label from the table. It should do the same thing as pressing the eye icon from the label dropdown.

![image](https://github.com/CATcher-org/WATcher/assets/87511888/0f8581a6-4644-493c-9a25-b6f76d27e133)
